pagopa/io-app
# chore(EU Covid Certificate): [IAGP-38] Handle message routing

## Short description
This pr handles the routing for different types of message. If the message is not an EU Covid Certificate message, navigate to `MESSAGE_DETAIL`, otherwise navigate to `EUCOVIDCERT_DETAILS`.

## List of changes proposed in this pull request
- Added `navigateToMessageRouterScreen` and replaced wherever it was used `navigateToMessageDetailScreenAction`, in order to add this new routing layer
- Changed `navigateToEuCovidCertificateDetailScreen` in order to accept also the `messageId` as parameter
- Changed `currentAuthCodeReducer` to `EuCovidCertCurrentSelectedState`, in order to store also the messageId associated with the AuthCode.
-  Added `MESSAGE_ROUTING_FAILURE` to track the reason of the message routing failure.

## How to test
- run `ts/screens/messages/__test__/MessageRouterScreen.test.tsx`
- Open a standard message, open a EU Covid message, should navigate to the right screen based on message type
- Open a message without loaded details, should display the `ErrorLoadingScreen`
